syntax = "proto3";

package microless.media.userreview;
option go_package = "microless/media/proto/userreview";

import "proto/data.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "google/api/annotations.proto";

message UploadUserReviewRequest {
    string user_id = 1;
    string review_id = 2;
    google.protobuf.Timestamp timestamp = 3;
}

message ReadUserReviewsRequest {
    string user_id = 1;
    int32 start = 2;
    int32 stop = 3;
}

message ReadUserReviewsRespond {
    repeated Review reviews = 1;
}

message CreateUserRequest {
    string user_id = 1;
}

service UserReviewService {
    rpc UploadUserReview (UploadUserReviewRequest) returns (google.protobuf.Empty);
    rpc ReadUserReviews (ReadUserReviewsRequest) returns (ReadUserReviewsRespond) {
        option (google.api.http) = {
            get: "/api/v1/userreview/{user_id}"
        };
    }
    rpc CreateUser (CreateUserRequest) returns (google.protobuf.Empty);
}